src/content/en/2020/resource-hints.md

```diff
@@ -5,19 +5,367 @@ chapter_number: 21
+  caption="The percent of preload hints on mobile using the scripts type.",
+  content="64%",
+  classes="big-number",
+  sheets_gid="1829901599",
+  sql_file="as_attribute_by_year.sql"
+) }}
+
+This is likely related to a large group of sites built as Single Page Apps that need the main bundle as soon as possible to start downloading the rest of their JS dependencies. Subsequent usage comes from font at 8%, style at 5%, image at 1%, and fetch at 1%.
+
+{{ figure_markup(
+  image="TODO",
+  caption="TODO",
+  description="TODO",
+  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vTYAbLxN40s6mNR1jo0XDe_V4siN8TAsx2mryMp5IQmlJ-9O9eJxYROz7Rw6ozyFP6hlIZHxxh95GqX/pubchart?oid=903180926&format=interactive",
+  sheets_gid="1829901599",
+  sql_file="as_attribute_by_year.sql"
+) }}
+
+Compared to the trend in 2019, we've seen rapid growth in font and style usage with the `as` attribute. This is likely related to both developers increasing the priority of critical CSS and also combining `preload` fonts with `display:optional` to [improve](https://web.dev/optimize-cls/#web-fonts-causing-foutfoit) Cumulative Layout Shift ([CLS](https://web.dev/cls/)).
+
+Be mindful that omitting the as attribute, or having an invalid value will make it harder for the browser to determine the correct priority and in some cases, such as scripts, can even cause the resource to be fetched twice.
+
+### The `crossorigin` attribute
+
+With `preload` and `preconnect` resources that have CORS enabled, such as fonts or images, it's important to include the `crossorigin` attribute, in order for the resource to be properly used. If the crossorigin attribute is absent, the request will follow the single-origin policy thereby making the use of preload useless.
+
+The latest trend indicates that by using `preload` with the `crossorigin` attribute we have 34% anonymous (or equivalent) cases, and 0.43% use-credentials cases. This has evolved in conjunction with the increase in font-preloading mentioned earlier.
+
+```html
+<link rel="preload" href="ComicSans.woff2" as="font" type="font/woff2" crossorigin>
+```
+
+Be mindful that fonts preloaded without the `crossorigin` attribute will be fetched [twice](https://web.dev/preload-critical-assets/#how-to-implement-relpreload)!
+
+### The `media` attribute
+
+When it's time to choose a resource for use with different screen sizes, reach for the `media` attribute with `preload` to optimize your media queries.
+
+```html
+<link rel="preload" href="desktop.css" as="style" media=”only screen and (max-width: 768px)”>
+<link rel="preload" href="mobile.css" as="style" media=”~~~”>
+```
+
+The most popular media value used with `preload` is a combination of `only screen and (max-width: 768px)` and `screen and (min-width: 767px)`. Seeing over 2,100 different combinations of media queries in the 2020 dataset encourages us to consider how wide the variance is between concept and implementation of responsive design from site to site.
```

Comment:
I'm not sure the most popular values are worth commenting on unless significantly more than the others? I was intrigued enough by this sentence [to look at the data](https://docs.google.com/spreadsheets/d/1lXjd8ogB7kYfG09eUdGYXUlrMjs4mq1Z7nNldQnvkVA/edit#gid=2016228012) and doesn't seem like these two values have any real significance:

![preload media values](https://user-images.githubusercontent.com/10931297/100557228-a2f3af00-329f-11eb-95d8-239138274bf2.png)

```suggestion
Seeing over 2,100 different combinations of media queries in the 2020 dataset encourages us to consider how wide the variance is between concept and implementation of responsive design from site to site.
```

Reply:
True, at the same time I find quite interesting the whole ecosystem is trying to conform on one particular group of number 768 and 767 to split desktop vs mobile.

I wonder how much the [breakpoints in w3schools](https://www.w3schools.com/css/css_rwd_mediaqueries.asp) contributed on this number.

What you think? Should we share something about this convergence or not?

Reply:
I'd say it's more likely to be due to the popularity of Bootstrap than w3schools!

But yes I think you can say something about this. Maybe the second sentence rather than the first will de-emphasise it. Something like:

```suggestion
Seeing over 2,100 different combinations of media queries in the 2020 dataset encourages us to consider how wide the variance is between concept and implementation of responsive design from site to site. The ever popular `767px/768px` breakpoints (as popularised by Bootstrap amongst others) can be seen in the data.
```